subworkflows/local/bam_infer_sex.nf

```diff
@@ -23,16 +24,15 @@ workflow BAM_INFER_SEX {
     ch_versions = ch_versions.mix(SOMALIER_EXTRACT.out.versions)
 
     SOMALIER_EXTRACT.out.extract
-        .map { meta, extract -> [ [ id: meta.project ], extract ] }
-        .groupTuple()
-        .join( ch_ped )
-        .set { ch_somalier_relate_in }
+        .combine( ch_ped.map { meta, ped -> ped } )
+        .filter { meta, extract, ped -> meta.sex == 0 }
```

Comment:
We frequently use a sex check as a sanity check to capture a potential sample mix-up (though it only works in 50% of the cases), or if the customer has simply but in the wrong sex in the order form. I understand why you don't want to run it for all the samples in some cases but it would be good to have the option to do so. For the cases where the sex is known there is no need to update the meta. We can simply parse the output of somalier after the pipeline has finished

Reply:
I still left the sex inference in RELATE_RELATE, and that is what is reported in multiQC, so the potential sample mix-up should still be caught (although without the pipeline doing a hard exit if such samples exists). 

Do you think that is reasonable? Or should the pipeline exit?

Reply:
Ah! I thought you removed it completely for samples that already had a known sex 🙃 This sounds like a good solution to me